nuxeo-ai-model/src/main/java/org/nuxeo/ai/cloud/NuxeoCloudClient.java

```diff
@@ -342,58 +314,64 @@ private AICorpus createCorpus(DocumentModel datasetDoc, String batch1, String ba
+                                                                     .setDocCount(trainingCount)
+                                                                     .setEvaluationDocCount(evalCount)
+                                                                     .setQuery(query)
+                                                                     .setSplit(split)
+                                                                     .setFields(fields)
+                                                                     .setTrainData(new AICorpus.Batch("0", batch1))
+                                                                     .setEvalData(new AICorpus.Batch("1", batch2))
+                                                                     .setStats(new AICorpus.Batch("2", batch3))
+                                                                     .setInfo(new AICorpus.Info(
+                                                                             dateFormat.format(start.toDate()),
+                                                                             dateFormat.format(end.toDate())))
+                                                                     .setJobId(jobId)
+                                                                     .setBatchId(batchId)
+                                                                     .build();
 
         return new AICorpus(jobId, props);
     }
 
+    @Override
+    public String predict(String modelName, TensorInstances instances) throws IOException {
+        Map<String, Serializable> params = new HashMap<>();
+        params.put(MODEL_NAME_PARAM, modelName);
+        params.put(DATASOURCE_PARAM, client.getConfiguration().getDatasource());
+        return getClient().api(API.Model.PREDICT).call(params, instances);
+    }
+
     @Override
     public JSONBlob getAllModels() throws IOException {
-        return getModels("models?properties=ai_model");
+        return getModels(API.Model.ALL, Collections.emptyMap());
     }
 
     @Override
     public JSONBlob getModelsByDatasource() throws IOException {
-        return getModels("models?properties=ai_model&datasource=" + datasource);
+        return getModels(API.Model.BY_DATASOURCE, Collections.emptyMap());
     }
 
     @Override
     public JSONBlob getPublishedModels() throws IOException {
-        return getModels("models?properties=ai_model&publishState=published&label=" + datasource);
+        String datasource = null;
+        if (getClient() != null) {
+            datasource = client.getConfiguration().getDatasource();
+        }
+        return getModels(API.Model.PUBLISHED, Collections.singletonMap(LABEL_PARAM, datasource));
     }
 
-    protected JSONBlob getModels(String parameters) throws IOException {
-        if (!isReady()) {
+    protected JSONBlob getModels(API.Model endpoint, Map<String, Serializable> params) throws IOException {
```

Comment:
Is it not something to put inside of the insight SDK/client all this part?

Reply:
I would like to keep it as a helper method. I believe the sdk has clear method of calling endpoints